grpc/interceptors.go

```diff
@@ -166,7 +166,7 @@ func (ci *clientInterceptor) intercept(
 	defer cancel()
 	// Disable fail-fast so RPCs will retry until deadline, even if all backends
 	// are down.
-	opts = append(opts, grpc.FailFast(false))
+	opts = append(opts, grpc.WaitForReady(true))
```

Comment:
Reference:

https://pkg.go.dev/google.golang.org/grpc#FailFast

> FailFast is the opposite of WaitForReady.
> Deprecated: use WaitForReady.

https://pkg.go.dev/google.golang.org/grpc#WaitForReady

> WaitForReady configures the action to take when an RPC is attempted on broken connections or unreachable servers. If waitForReady is false and the connection is in the TRANSIENT_FAILURE state, the RPC will fail immediately. Otherwise, the RPC client will block the call until a connection is available (or the call is canceled or times out) and will retry the call if it fails due to a transient error. gRPC will not retry if data was written to the wire unless the server indicates it did not process the data. Please refer to https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md.

> By default, RPCs don't "wait for ready".